src/FiberTraces.cc

```diff
@@ -1560,9 +1561,21 @@ namespace pfsDRPStella = pfs::drp::stella;
                                                                                                            fiberTraceFunctionFindingControl );
         if (findCenterPositionsOneTraceResult.apertureCenterIndex.size() > fiberTraceFunctionFindingControl->minLength){
           B_ApertureFound = true;
-          const ndarray::Array<double, 1, 1> D_A1_ApertureCenterIndex = ndarray::external(findCenterPositionsOneTraceResult.apertureCenterIndex.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.apertureCenterIndex.size())), ndarray::makeVector(1));
-          const ndarray::Array<double, 1, 1> D_A1_ApertureCenterPos = ndarray::external(findCenterPositionsOneTraceResult.apertureCenterPos.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.apertureCenterPos.size())), ndarray::makeVector(1));
-          const ndarray::Array<double, 1, 1> D_A1_EApertureCenterPos = ndarray::external(findCenterPositionsOneTraceResult.eApertureCenterPos.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.eApertureCenterPos.size())), ndarray::makeVector(1));
+          const ndarray::Array<float, 1, 1> F_A1_ApertureCenterIndex = ndarray::external(findCenterPositionsOneTraceResult.apertureCenterIndex.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.apertureCenterIndex.size())), ndarray::makeVector(1));
+          const ndarray::Array<float, 1, 1> F_A1_ApertureCenterPos = ndarray::external(findCenterPositionsOneTraceResult.apertureCenterPos.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.apertureCenterPos.size())), ndarray::makeVector(1));
+          const ndarray::Array<float, 1, 1> F_A1_EApertureCenterPos = ndarray::external(findCenterPositionsOneTraceResult.eApertureCenterPos.data(), ndarray::makeVector(int(findCenterPositionsOneTraceResult.eApertureCenterPos.size())), ndarray::makeVector(1));
+          ndarray::Array<double, 1, 1> D_A1_ApertureCenterIndex = ndarray::allocate(F_A1_ApertureCenterIndex.getShape()[0]);
+          auto itDI = D_A1_ApertureCenterIndex.begin();
+          for (auto itF = F_A1_ApertureCenterIndex.begin(); itF != F_A1_ApertureCenterIndex.end(); ++itF, ++itDI)
+            *itDI = double(*itF);
+          ndarray::Array<double, 1, 1> D_A1_ApertureCenterPos = ndarray::allocate(F_A1_ApertureCenterPos.getShape()[0]);
+          auto itDP = D_A1_ApertureCenterPos.begin();
+          for (auto itF = F_A1_ApertureCenterPos.begin(); itF != F_A1_ApertureCenterPos.end(); ++itF, ++itDP)
+            *itDP = double(*itF);
+          ndarray::Array<double, 1, 1> D_A1_EApertureCenterPos = ndarray::allocate(F_A1_EApertureCenterPos.getShape()[0]);
+          auto itDE = D_A1_EApertureCenterPos.begin();
+          for (auto itF = F_A1_EApertureCenterPos.begin(); itF != F_A1_EApertureCenterPos.end(); ++itF, ++itDE)
+            *itDE = double(*itF);
```

Comment:
There's a lot of repeated code here.  Can you factor it out into a function?  Actually, I think ndarray should be able to do this with its copy constructor.
You should have braces around the body of the `for` loop so it doesn't get lost accidentally.